### PR TITLE
fix(docs): tokenize footer styles

### LINKED ISSUE #4061: Review footer.css styles for replacing custom styles with existing styling options or tokenization
We should evaluate these styles to determine if they're still needed and remove them if not.

If they are still needed we should update the currently commented out global styles with their proper token replacements.

### PR DESCRIPTION
**Pull request by evwilkin**
Closes #4061 

This PR:
- Replaces custom styles in footer.js with tokens, or removes custom styles completely where possible
- Fixes GDPR banner functionality (currently it always shows, is not dismissable)

### CODE REVIEW

**Review by kmcfaul — APPROVED** [review]
Looks great! Glad to see some of the custom styling cleaned up!

File: packages/documentation-framework/components/footer/footer.css [hunk 1]
```diff
@@ -1,116 +1,53 @@
...
-/*   font-family: var(--pf-v6-global--FontFamily--text); */
-  font-weight: 300;
-  background-color: #1a1a1a !important;
-  --pf-v6-c-page__main-section--PaddingTop: var(--pf-t--global--spacer--xl);
-  --pf-v6-c-page__main-section--PaddingRight: var(--pf-t--global--spacer--2xl);
-  --pf-v6-c-page__main-section--PaddingBottom: var(--pf-t--global--spacer--2xl);
-  --pf-v6-c-page__main-section--PaddingLeft: var(--pf-t--global--spacer--2xl);
+.ws-org-pfsite-l-footer,
+.ws-org-pfsite-l-footer-dark {
+  color: var(--pf-t--global--text--color--300);
```
Comment (nicolethoen) [inline packages/documentation-framework/components/footer/footer.css:3]: can we use a sematic token here instead?

File: packages/documentation-framework/components/footer/footer.css [hunk 2]
```diff
@@ -1,116 +1,53 @@
...
+  background-color: var(--pf-t--color--gray--95);
```
Comment (nicolethoen) [inline packages/documentation-framework/components/footer/footer.css:4]: and here?

File: packages/documentation-framework/components/footer/footer.css [hunk 3]
```diff
@@ -1,116 +1,53 @@
...
-  /* Hide long overflowing tocs */
-  z-index: 1;
-}
-.pf-v6-c-page__main-section.ws-org-pfsite-l-footer-dark > * {
-  font-weight: 300;
-  font-size: 12px;
-  color: #d2d2d2;
+/* Red Hat footer */
+.ws-org-pfsite-l-footer-dark > * {
+  font-size: var(--pf-t--global--font--size--100);
```
Comment (nicolethoen) [inline packages/documentation-framework/components/footer/footer.css:41]: not a semantic token

**Review by nicolethoen — COMMENTED** [review]
I am not sure if a design issue ended up open to address footer styles. I'd love some design feedback on the footer ahead of final release. It could be a follow up issue to this, though.

I tagged @bekah-stephens for her input

**Review by bekah-stephens — CHANGES_REQUESTED** [review]
Changes are looking good so far! Since we're updating tokens to use our new semantic versions, this would be a good chance to update the light theme version of the footer as well. Screenshot & token suggestions below!

image


1. footer (main section background) use _**--global--background--color--secondary--default**_
2. footer-menu-list-item use _**--global--text--color--subtle**_
3. footer-menu-link use _**--global--text--color--link--default**_
4. footer-column (box background) use _**--global--background--color--primary--default**_
5. update the PF logo to be the same coloring as in the masthead
6. footer-menu-about-description use _**--global--text--color--regular**_
7. footer-menu-social-link use _**--global--icon--color--brand--default**_

tagging @lboehling to double check my token suggestions

**Comment by evwilkin** [comment]
Thanks @bekah-stephens your feedback was super helpful to get this corrected, hopefully everything matches what you suggested now!

**Review by bekah-stephens — APPROVED** [review]

**Review by bekah-stephens — CHANGES_REQUESTED** [review]
A few minor things:

- The footer menu list titles "Quicklinks" and "Contribute" are using the _**"Red Hat Display"**_ font family, while the "Stay in touch" title is using _**"Red Hat Text"**_. They should all be using **_"Red Hat Text"_**
- We're moving away from using all caps, so those three titles should be **_sentence case_** so that we're consistent with the design language type guidelines
- For all three titles use a **_"semibold" font-weight_** rather than "bold"

**Comment by evwilkin** [comment]
Thanks @bekah-stephens & @lboehling - updated now with those changes 👍

**Review by bekah-stephens — APPROVED** [review]
Looks good!

**Pull request merged**